### ISSUE TITLE
Dependency for redis 2.8.0+ with support for protocol rediss:// (TLS)

### ISSUE THREAD

**Issue by autorTepatiki** [jpodwys/cache-service-redis#21]
Hi @jpodwys , I would to thank you for the job in this package. In this direction, I wish to have the latest changes made to "node_redis" v2.8.0+ in the past weeks, supportting "rediss://" protocol usng TLS. It is already included in the master branch of https://github.com/NodeRedis/node_redis , but for some reason it is not been included in your distribution. Maybe you can push some update in order to get this? Also a funtional test about rediss:// would be fine. Regards.

**Comment by jpodwys** [comment]
Hi, I'm glad you're finding the library useful!

If I understand you correctly, you'd like `cache-service-redis` to depend on `node_redis@2.8.0`. This package's dependency on `node_redis` is a relative pin: `"redis": "^2.6.2"`. I just installed it to be certain and it does indeed install `node_redis@2.8.0`.

If you're not seeing that, then please simply update your installation. Please let me know if I've misunderstood.

**Comment by autorTepatiki** [comment]
Hi. As the package "redis" is not been tagged since 2 years ago, it doesnt include the rediss:// protocol nor TLS, fundamental pieces to connect to ElasticCache. My question tried to address this point, trying to see if the dependency in your package could be changed directly to an own fork from Github (master branch of "redis") to get those changes reflected. This would be "A" plan, because the other library that really supports rediss:// (TLS) is "ioredis", but this will be a mayor change in your repo I think.

**Comment by jpodwys** [comment]
Thank you for helping me to understand, I didn't realize that there hasn't been a release in over two years.

I'm uncomfortable pinning the master branch of `cache-service-redis` to the master branch of `node_redis`.

Here are two possible solutions:

1. I'd like to cut a new major release (`3.0.0`) that would switch to `ioredis`. I'm not certain how much work is involved, but I just switched to `ioredis` locally and all of my unit tests still pass. I'm guessing it breaks things somewhere, but I need to investigate more.
2. If you can't wait for/contribute to such a release, then please feel free to fork both repos and pin to your forks in the mean time.

What do you think?

**Comment by autorTepatiki** [comment]
Yes, that is. Both options are fine to me, Plan 1 + plan 2. Thank you very much.

**Comment by jpodwys** [comment]
@martindcernadas I haven't yet had time to update this beyond running the unit tests with `ioredis`. I noticed that you made a fork and switched to `ioredis`. I'm curious if you've used it. If so, how is it functioning for you?
